### ISSUE TITLE
Scala plugin's scaladoc task broken with Scala 2.13.0

### ISSUE THREAD

**Issue by dubinsky** [gradle/gradle#9855]
### Expected Behavior
When `gradle scaladoc` is run, Scaladoc should be generated.

### Current Behavior
When using Scala 2.13.0:

```
  Execution failed for task ':scaladoc'.
  java.net.MalformedURLException
```

### Context
When publishing to JCenter, I include Scaladoc jar - which is impossible to generate with Scala 2.13.

### Steps to Reproduce 
In the attached project [gradle-scaladoc-bug.zip](https://github.com/gradle/gradle/files/3352979/gradle-scaladoc-bug.zip)

```
  $ gradle scaladoc -PscalaVersion=2.13.0
```

Reason:
  https://github.com/gradle/gradle/blob/master/subprojects/scala/src/main/java/org/gradle/api/internal/tasks/scala/AntScalaDoc.java#L53 assumes presence of scala/tools/ant/antlib.xml, which was removed:   https://github.com/scala/scala/pull/6255; see also  https://users.scala-lang.org/t/scala-2-13-0-ant-support/4694/11

### Your Environment
Build scan URL: https://scans.gradle.com/s/7wmsskmoua3uu

**Comment by barrybecker4** [comment]
I have encountered the same problem using Scale 13.1 and gradle 4.10.2

**Comment by ljacomet** [comment]
Gradle 6.0 will have improvements, including better support of Scala 2.13.

When [6.0 RC1 is out](https://github.com/gradle/gradle/milestone/83), could you test again?

**Comment by dubinsky** [comment]
@ljacomet :
> Gradle 6.0 will have improvements, including better support of Scala 2.13.
> When [6.0 RC1 is out](https://github.com/gradle/gradle/milestone/83), could you test again?

I did; it breaks the same way. "Better support for Scala 2.13" is not going to fix this issue; Gradle Scala plugin needs to be reworked to NOT use `scala/tools/ant/antlib.xml` (removed in Scala 2.13.0) to generate Scaladoc.

**Comment by lacasseio** [comment]
Scala 2.13 indeed broke everyone by removing the Ant support. We looked into fixing the issue by executing the work directly inside a worker action (https://github.com/gradle/gradle/tree/lacasseio/scala-doc-fix-for-2.13) however we hit an issue with the classloader which makes it impossible to run the action hence doesn't fix the issue.

At the moment, it would be recommended to build https://github.com/cigaly/scala-ant and adding that JAR to the `ScalaDoc` task's `scalaClasspath` property. We didn't confirm nor wrote a sample for this workaround, however, it should be the path forward to unblock users immediately.

We will look into the classloader issue with the worker API to see what needs to be fixed so we can move forward with a proper fix to handle 2.13 breakage.

**Comment by dubinsky** [comment]
> We will look into the classloader issue with the worker API to see what needs to be fixed so we can move forward with a proper fix to handle 2.13 breakage.

Thank you!!!

**Comment by SimY4** [comment]
Just if anyone is interested. Minimal gradle build configuration that worked for me:

```
configurations {
    scaladocExtras
}

dependencies {
    scaladocExtras 'org.scala-lang:scala-library:2.13.1'
    scaladocExtras 'org.scala-lang:scala-compiler:2.13.1'
    scaladocExtras 'org.scala-lang:scala-reflect:2.13.1'
    scaladocExtras 'org.apache.ant:ant:1.9.14'
}

scaladoc {
    scalaClasspath = configurations.scaladocExtras + files("lib/scala-ant-2.13.1.jar")
}
```

**Comment by barrybecker4** [comment]
Just to be clear, it looks like this should be fixed in release 6.2 of Gradle. Please correct me if that is not correct.

**Comment by big-guy** [comment]
Yep, this is fixed in release and will be in 6.2
